designs/2023-rule-options-defaults/README.md

```diff
@@ -30,144 +30,100 @@ For example:
-| --------------------------------------------- | ------------------------------------ | -------------------------------------- |
-| `"error"`                                     | `[]`                                 | `[ { enforceForClassFields: true } ]`  |
-| `["error"]`                                   | `[]`                                 | `[ { enforceForClassFields: true } ]`  |
-| `["error", {}]`                               | `[{ enforceForClassFields: true }]`  | `[ { enforceForClassFields: true } ]`  |
-| `["error", { enforceForClassFields: false }]` | `[{ enforceForClassFields: false }]` | `[ { enforceForClassFields: false } ]` |
-| `["error", { enforceForClassFields: true }]`  | `[{ enforceForClassFields: true }]`  | `[ { enforceForClassFields: true } ]`  |
+- `strict`: The mode depends on -and can be overridden based on- `ecmaFeatures.impliedStrict`
 
-### Implementation
+</details>
 
-Currently, ESLint rule options are passed through a [`getRuleOptions` function](https://github.com/eslint/eslint/blob/da09f4e641141f585ef611c6e9d63d4331054706/lib/linter/linter.js#L740) whose only processing is to remove severity.
-That function could be augmented to also take in the `rule.meta.schema`.
-Its "worker" logic could recursively parse the rule config and schema:
+### Schema Defaults
 
-1. If the rule config is `null`, return that directly
-2. If the schema is an array:
-   1. Recurse on the range of elements from 0 to the smaller of the rule config and schema's length
-   2. If there are remaining rule config values (so, there were more rule configs than schema descriptions), add them directly to the resolved options
-   3. If there are remaining schema (so, there were more schema descriptions than rule configs), fill in any options values for schemas containining either `default` or `type: "object"`
-3. Create an options object that is a shallow clone of the first non-`undefined` from `ruleConfig` or `schema.default`, or `{}` if the schema contains `type: "object"`
-4. If the options is truthy and schema is an object:
-   1. Recursively apply defaulting logic for each of the schema's `properties`
+Ajv provides its own support for `default` values with a [`useDefaults` option](https://ajv.js.org/guide/modifying-data.html#assigning-defaults).
+It's used in ESLint core already, in [`lib/shared/ajv.js`](https://github.com/eslint/eslint/blob/22a558228ff98f478fa308c9ecde361acc4caf20/lib/shared/ajv.js#L21).
+Some core ESLint rules describe `default:` values in that schema.
 
-Draft PR: [[Reference] feat!: factor in schema defaults for rule options](https://github.com/eslint/eslint/pull/17580)
+Having two places to describe `default` values can be confusing for developers.
+It can be unclear where a value comes from when its default may be specified in two separate locations.
+This RFC proposes removing the `useDefaults: true` setting in the next major version of ESLint.
```

Comment:
We'd probably end up doing this in v10, which would give us time to transition folks over.